lib/course/dsl/routing.rb

```diff
@@ -0,0 +1,81 @@
+module Course
+  module DSL
+    module Routing
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      def call(env)
+        return self.class.find_get_endpoint(env).call if handle_get_request?(env)
+        return self.class.find_post_endpoint(env).call if handle_post_request?(env)
+        return self.class.find_rack_app(env).new.call(env) if handle_other_rack_request?(env)
+
+        [404, {}, [""]]
```

Comment:
it should raise an error, but not 404 (not found), 
500 would be better in this case